hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/execution/StoreSnapshotTasklet.java

```diff
@@ -165,4 +183,24 @@ public String toString() {
         /** Input is done, terminal state. */
         DONE
     }
+
+    private static class Metrics implements DynamicMetricsProvider {
+
+        private long bytes;
+        private long keys;
+
+        synchronized void set(long bytes, long keys) {
+            this.bytes = bytes;
+            this.keys = keys;
+        }
+
+        @Override
+        public synchronized void provideDynamicMetrics(MetricDescriptor descriptor, MetricsCollectionContext context) {
+            if (bytes > 0) {
```

Comment:
I don't think we need the conditional logic here. It'll be confusing if metric is sometimes missing and sometimes not.

Reply:
Ok, I can remove it, it's just that for most vertices there are no snapshots, ever, so there are a lot of zeroes we publish (and I have not found a better way to tell about a vertex if that's the case or not). Anyways, will remove.